liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/ApplicationXmlDocument.java

```diff
@@ -52,12 +52,16 @@ public void createElement(String element, String attrKey, String attrValue) {
     }
     
     public void writeApplicationXmlDocument(File serverDirectory) throws IOException, TransformerException {
-        File configDropins = new File(serverDirectory, "configDropins/defaults");
-        configDropins.mkdirs();
-        File applicationXml = new File(configDropins, APP_XML_FILENAME);
-        if (hasChildElements()) {
-            writeXMLDocument(applicationXml);
+        File applicationXml = getApplicationXmlFile(serverDirectory);
+        if (!applicationXml.getParentFile().exists()) {
+            applicationXml.getParentFile().mkdirs();
         }
+        writeXMLDocument(applicationXml);
+    }
+    
+    public File getApplicationXmlFile(File serverDirectory) {
+        File f = new File(serverDirectory, "configDropins/defaults/" + APP_XML_FILENAME); 
```

Comment:
how about using the goal name and call it install_apps_configuration_1491924271.xml